aws/src/integration/java/org/apache/iceberg/aws/s3/TestS3FileIOIntegration.java

```diff
@@ -268,4 +306,11 @@ private void validateRead(S3FileIO s3FileIO) throws Exception {
     stream.close();
     Assert.assertEquals(content, result);
   }
+
+  private String testAccessPointARN() {
```

Comment:
I think we also need to test cross-region endpoint, because I see this S3 client config:

```
    /**
     * Returns whether the client is allowed to make cross-region calls when an S3 Access Point ARN has a different
     * region to the one configured on the client.
     * <p>
     * @return True if a different region in the ARN can be used.
     */
    public boolean useArnRegionEnabled() {
        return useArnRegionEnabled.value();
    }
```

Maybe we should turn this on when access point is configured. Could you check?

Reply:
- by configured you mean there is a s3bucketToAccessPoint is not empty or introducing a flag to enable this feature.
- From the looks of this, it looks like it would be required trying out a quick test case for the same.

Reply:
I tried this and it failed with "java.lang.IllegalArgumentException: The region field of the ARN being passed as a bucket parameter to an S3 operation does not match the region the client was configured with. To enable this behavior and prevent this exception set 'useArnRegionEnabled' to true in the configuration when building the S3 client. Provided region: 'us-east-2'; client region: 'us-east-1'."

Looks like we need to set this

Reply:
cool, so to summarize, same-single-region and multi-region don't need this flag, and cross-single-region need this flag, is that right?

If so, I think we should introduce this flag, because there is some performance implications in enabling this feature as the signing region is changed. We should not enable this feature by default for people who use MRAP.

Reply:
+1, same-single-region / multi-region doesn't need this (verified this locally)

should we make this flag explicit to access points`s3-acess-point.cross-single-region.enabled` or it can be as general as `s3-cross-single-region.enabled` based on that we can set this in our s3 client the required conf.

Reply:
For boolean config, prefer to use `-enabled` suffix as convention. All S3 features should be under `s3.` config namespace.

I think we can just follow the S3 variable name, `s3.arn-region-enabled` since S3 already gave this feature this name. We can directly link to this feature's github link in AWS SDK v2 repo as a reference. What do you think?

Reply:
This makes sense, having such a name and linking it to the direct AWS SDK v2 repo as a reference adds more value. Let me make the changes.